crates/matrix-sdk/src/sliding_sync.rs

```diff
@@ -148,7 +146,7 @@ pub struct SlidingSyncRoom {
     is_loading_more: Mutable<bool>,
     is_cold: Arc<AtomicBool>,
     prev_batch: Mutable<Option<String>>,
-    timeline: AliveRoomTimeline,
+    timeline: Vec<SyncTimelineEvent>,
```

Comment:
But this will now clone a lot of data when we clone the `SlidingSyncRoom`, are we sure we don't need to have this behind an `Arc`?

Reply:
I'm not sure we are cloning this `SlidingSyncRoom` quite often actually. cc @gnunicorn?

Reply:
Well it's not up to us if we clone it or now. We have a public getter which clones the room:

https://github.com/matrix-org/matrix-rust-sdk/blob/c7fe1ddffbf15d502efda97b3bda00501f89afd2/crates/matrix-sdk/src/sliding_sync.rs#L816-L819

So it totally depends on what the user does, but I can imagine users of the SDK having clones of the room in a bunch of places. Furthermore if we now remove the `Arc`, users that picked up a room using the getter won't get changes to the `timeline` field anymore. 

If they then call the `timeline()` method using a `SlidingSyncRoom` that has such stale data, they will get a timeline that has been populated with these stale events.